Automattic/pocket-casts-android
# Remove ADD_PATRON_ENABLED feature flag

## Description
- This is one more feature flag removal. This time was `ADD_PATRON_ENABLED` feature flag
- This PR also removes code related to the old subscription model

Closes #1675

## Testing Instructions
1. Apply this [upgrade_plan.patch](https://github.com/Automattic/pocket-casts-android/files/13876198/upgrade_plan.patch) to smoke test this feature
2. Log in with an user without subscription plan
3. Go to Profile
4. Go to account
5. Make sure you see the carrousel with Plus plan and Patron plan
6. Log in with a patron user
7.  Go to Profile
8. Make sure you see patron set in your profile
9. Go to account
10. Make sure you see the patron details with the `Renew your subscription` button 

Feel free to do more smoke tests

## Screenshots
| Free | Patron |
|--------|--------|
|![Screenshot_20240109-131654](https://github.com/Automattic/pocket-casts-android/assets/42220351/10888042-3437-408e-95e2-e0d0c15de43a) |![Screenshot_20240109-132120](https://github.com/Automattic/pocket-casts-android/assets/42220351/82c9db49-b307-41f4-809a-d4980d7009f1) | 



## Checklist
 
#### I have tested any UI changes...
